### PR TITLE
`ArrayInput`: Accept non-array input without throwing an exception

### LINKED ISSUE #104: Array Input has strange behaviour
I would expect that using an `ArrayInput` with `InputFilter` to fail validation when given non-array input, but instead an exception is thrown.

Given that the whole idea is filtering and validation of untrusted input, causing a crash in this situation seems odd to me.

```php
$inputFilter = new InputFilter();
$inputFilter->add([
    'type'       => ArrayInput::class,
    'validators' => [
        ['name' => NotEmptyValidator::class],
    ],
], 'myInput');

$inputFilter->setData(['myInput' => 'Whatever']); // Exception thrown here.
assert($inputFilter->isValid() === false);
```

### PR DESCRIPTION
**Pull request by gsteel**
### Description

Alters the behaviour of `ArrayInput` so that non-array input is deemed a validation failure instead of an exceptional condition.

Fixes #104

This could be considered a BC break, but it would _never_ have previously been possible for a non-array value to be validated/filtered here, so it's more likely that no-one ever used this input type without wrapping its execution in a try/catch block.

Assuming users are using this input type, and they _are_ wrapping with try, then the introduction of the IsArray validator will still provide graceful validation failure.

### CODE REVIEW

File: src/ArrayInput.php [hunk 1]
```diff
@@ -4,60 +4,30 @@
 
 namespace Laminas\InputFilter;
 
-use function gettype;
+use Laminas\Validator\IsArray;
```
Comment (Ocramius) [inline src/ArrayInput.php:7]: Fancy new tech :P

**Review by Ocramius — APPROVED** [review]
LGTM :+

**Comment by Ocramius** [comment]
Marked as bug + improvement on purpose, BTW: it changes dependencies, so it can't go to a patch version

**Comment by gsteel** [comment]
Thankyou @Ocramius 🙏

File: src/ArrayInput.php [hunk 2]
```diff
@@ -4,60 +4,30 @@
...
         }
-        parent::setValue($value);
-
-        return $this;
-    }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function resetValue()
```
Comment (kersysgediminas) [inline src/ArrayInput.php:37]: @gsteel After removing this, default value is null now instead of empty array. It looks like a breaking change.

**Pull request merged**